src/design/components/organisms/Modal/index.tsx

```diff
@@ -223,6 +232,30 @@ const ModalItem = ({
     [closeModal]
   )
 
+  useEffectOnce(() => {
+    if (modal.navigation != null) {
+      push(modal.navigation.url)
+    }
+  })
+
+  useEffect(() => {
+    return () => {
+      willUnmountRef.current = true
+    }
+  }, [])
+
+  useEffect(() => {
+    return () => {
+      if (willUnmountRef.current && modal.navigation != null) {
```

Comment:
Difficult to understand what is going on. Would be better to extract this hook with a descriptive function name.
Something like the below.

```ts
const Modal = () => {
  ...
  useFallbackNavigation(modal.navigation)
  ...
}


function useFallbackNavigation(navigation: {...}) {
  const { ... } = useRouter()
  useEffect(() => { ... })
}
```